projects/packages/publicize/src/class-publicize-base.php

```diff
@@ -2100,6 +2100,48 @@ public function get_dismissed_notices() {
 	public static function can_manage_connection( $connection_data ) {
 		return current_user_can( 'edit_others_posts' ) || get_current_user_id() === (int) $connection_data['user_id'];
 	}
+
+	/**
+	 * Gets the share status for a post.
+	 *
+	 * @param int $post_id The post ID.
+	 */
+	public function get_post_share_status( $post_id ) {
```

Comment:
Can we avoid dumping this into the base class? We can have a dedicated class for share status, just like we have for [connections](https://github.com/Automattic/jetpack/blob/1794810ac6f4f89e2bcda04d13583dbf65d73760/projects/packages/publicize/src/class-connections.php), [services](https://github.com/Automattic/jetpack/blob/1794810ac6f4f89e2bcda04d13583dbf65d73760/projects/packages/publicize/src/class-services.php), etc.

Reply:
6b39a7e